js/fetch.ts

```diff
@@ -410,28 +420,58 @@ export async function fetch(
     }
   }
 
-  // Send Fetch message
-  const builder = flatbuffers.createBuilder();
-  const headerOff = msgHttpRequest(builder, url, method, headers);
-  const resBase = await sendAsync(
-    builder,
-    msg.Any.Fetch,
-    msg.Fetch.createFetch(builder, headerOff),
-    body
-  );
-
-  // Decode FetchRes
-  assert(msg.Any.FetchRes === resBase.innerType());
-  const inner = new msg.FetchRes();
-  assert(resBase.inner(inner) != null);
-
-  const header = inner.header()!;
-  const bodyRid = inner.bodyRid();
-  assert(!header.isRequest());
-  const status = header.status();
-
-  const headersList = deserializeHeaderFields(header);
-
-  const response = new Response(status, headersList, bodyRid);
-  return response;
+  while (true) {
+    // Send Fetch message
+    const builder = flatbuffers.createBuilder();
+    const headerOff = msgHttpRequest(builder, url, method, headers);
+    const resBase = await sendAsync(
+      builder,
+      msg.Any.Fetch,
+      msg.Fetch.createFetch(builder, headerOff),
+      body
+    );
```

Comment:
This code which builds the flatbuffer message, calls sendAsync(), and deserializes the response should be in its own function.